### PR TITLE
New RavenDB, Azure, Dyn Templates & Updated Amazon Template

### PR DESCRIPTION
**Pull request by timhunt303**
A mixture of step templates, with 3 completely new ones and one that has been updated.

### CODE REVIEW

File: step-templates/azure-upload-files.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Name": "storageAccountKey",
+      "Label": "Azure Storage Account Key",
+      "HelpText": "The key that is used to log into the account.",
+      "DefaultValue": null,
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
```
Comment (alfhenrik) [inline step-templates/azure-upload-files.json]: This should be a Sensitive/Password control

**Comment by alfhenrik** [comment]
In the script body

```
...
#create new A record

if($createNewARecord -eq $TRUE)
{
    $dynCreateURI = "https://api.dynect.net/REST/ARecord/fewzion.com/dummy.fewzion.com"   
    ...
}
...
```
That URL should probably not point to that domain

**Comment by alfhenrik** [comment]
Other than those two comments, it all looks good

**Comment by timhunt303** [comment]
@alfhenrik changes have been made

**Comment by alfhenrik** [comment]
Thanks heaps @timhunt303 !

**Pull request merged**